conda/core/package_cache.py

```diff
@@ -412,8 +440,21 @@ def make_actions_for_dist(dist, record):
             )
             return None, extract_axn
 
+        # At this point we know we need an empty package cache slot for a tarball
+        # and an extracted dist, so find the first such slot. First fine one that
+        # will not require deleting or overwriting an existing cache entry
+        target_filename = dist.to_filename()
+        target_dirname = dist.dist_name
+        first_writable_cache = first(
```

Comment:
If we want to be more aggressive about preferring the first writable package cache, we'd do this instead:
```
first_writable_cache = PackageCache.first_writable()
if first_writable_cache.contains_file(target_filename) or first_writable.contains_file(target_dirname):
    first_writable_cache = first_writable_cache.new_subdir()
```
Honestly, now that I've typed it, I prefer it.